server/mscalendar/oauth2.go

```diff
@@ -123,9 +123,15 @@ func (app *oauth2App) CompleteOAuth2(authedUserID, code, state string) error {
 
 	if mattermostUserID == app.Config.BotUserID {
 		app.Poster.DM(authedUserID, BotWelcomeMessage, me.Mail)
-	} else {
-		app.Poster.DM(mattermostUserID, WelcomeMessage, me.Mail)
+		return nil
 	}
 
+	err = app.Store.StoreUserInIndex(u)
```

Comment:
Just to make sure: we do not want the bot user to be in the user index? It seems that we used to have it there (I might be wrong).

Reply:
Yes, I am kind of doing a drive-by refactor that makes it so we don't have to access the `userIndex` twice during the status sync job. The first access is usually spent filtering out the bot user, but if it is not in the index to begin with (which is want we originally wanted to do), then we do not need to filter it out.

Reply:
To be clear, this sort of change would not be acceptable if we had already released the plugin. But since it is still in development, it is okay to change different assumptions about how the store is used.